sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2970,9 +2971,14 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
   /**
    * Create an [[UnresolvedRelation]] from an identifier reference.
    */
-  private def createUnresolvedRelation(
-      ctx: IdentifierReferenceContext): LogicalPlan = withOrigin(ctx) {
-    withIdentClause(ctx, UnresolvedRelation(_))
+  private def createUnresolvedRelation(identifier: IdentifierReferenceContext,
+      optionsClause: Option[OptionsClauseContext] = None): LogicalPlan =
+    withOrigin(identifier) {
+      val options = optionsClause.map{clause =>
```

Comment:
nit: we need space before and after `{`
`optionsClause.map{clause` should be `optionsClause.map { clause`